typegate/src/runtimes/typegraph.ts

```diff
@@ -158,11 +170,29 @@ export class TypeGraphRuntime extends Runtime {
         };
 
         visitTypes(this.tg, getChildTypes(this.tg.types[0]), myVisitor);
-
-        const scalarTypes = distinctBy(
-          [...scalarTypeIndices].map((idx) => this.tg.types[idx]),
+        const distinctScalars = distinctBy(
+          [...scalarTypeIndices].map((idx) => {
+            const node = this.tg.types[idx];
+            this.scalarIndex.set(node.type, idx);
+            return node;
+          }),
           (t) => t.type, // for scalars: one GraphQL type per `type` not `title`
-        ).map((type) => this.formatType(type, false, false));
+        );
+        const scalarTypes = distinctScalars.map((type) =>
+          this.formatType(type, false, false)
+        );
+        const customScalarTypes = distinctScalars.map((node) => {
+          const key = generateCustomScalarName(node);
+          const custom = {
+            title: key,
+            type: "object",
+            properties: {
+              [node.type]: this.scalarIndex.get(node.type),
+            },
+          } as ObjectNode;
+          this.customScalars.set(key, custom);
```

Comment:
This custom does not seem to at "the correct place", why not generating this payload later in the formatType function? This would avoid to "hack" a class attribute for that.

Reply:
I was projecting on potentially re-use the types, it was also motivated by explicitly throwing a custom error in case it does not exist in the map.

Reply:
I would advise against keeping something in memory for a potential re-use (by another subsequent query).

When there is a class attribute, the user expect that this attributes are global to the "work scope" of the class, which is not the case: only when the materializer is "schema", this field is used (e.g. the field could be passed as an argument to the formatType function). Or better, the `custom` object creation can be done directly in formatType, avoiding having to keep in memory some we have prebuilt (and maybe that will not be used if the query do not ask for those fields).

Reply:
I still kept `scalarIndex` class property because it would require passing a scalarIndex to every `formatType` call, thus adding more complexity and more branching and more `undefined`s checks in case it is not used. `formatType` is just as useful as a helper function in a class in this context. The content is also cleared when all calls are terminated anyway.